### ISSUE TITLE
Clear chat buffer on join

### ISSUE THREAD

**Issue by hecktest** [minetest/minetest#9225]
If you're developing a mod, and you happen to get a lua crash (such as due to a typo) and then quickly reload the game, the old error will still be dumped to the chat.
![error dump](https://user-images.githubusercontent.com/42101236/71259765-76b67b00-2339-11ea-9c62-6cbc092f3703.png)
It would be nice if this did not happen; a clean state is preferrable when you're working on anything visual, and any scary red error dumps should be related to the current session; users should not be conditioned to ignore those either.

**Comment by Wuzzy2** [comment]
I 100% agree, this is really confusing.

**Comment by fluxionary** [comment]
I don't disagree that seeing old errors is confusing, but i certainly make use of the old chat buffer if i need to relog for some reason.

**Comment by rubenwardy** [comment]
This could break the chat resend feature if done without care, I often reload the server and send the same chat message whilst testing

**Comment by LoneWolfHT** [comment]
I'd be happy if we at least got something to tell us we're looking at an old chat

**Comment by SmallJoker** [comment]
Problem:
https://github.com/minetest/minetest/blob/4b9fabf172dafa325a2adb9754c581a74d0f4e7b/src/client/game.cpp#L2818

This static variable is destructed only when closing Minetest. It could be moved somewhere else (global?) to flush the buffers when leaving the world.

**Comment by Wuzzy2** [comment]
> This could break the chat resend feature if done without care, I often reload the server and send the same chat message whilst testing

I don't really know what you mean by this. (insert confused face here)

**Comment by ClobberXD** [comment]
While I'd like to see the chat cleared, I'd prefer to have my input history (the list of previously-entered inputs scroll-able using arrow keys) intact. I hope the two aren't linked together.

**Comment by hecktest** [comment]
@Wuzzy2 You can press the up arrow in chat to re-use a line you've previously typed; the history persists between sessions (though not between processes). It's a useful thing.

A nondestructive way to solve this would be something like the "clear screen" escape that terminals use. (not literally clearing the screen, just marking a barrier)

**Comment by Wuzzy2** [comment]
I agree that the chat history should not be cleared, only the visible chat on start.

**Comment by hecktest** [comment]
Just noticed that closing the server cleanly and reopening it makes the error go away. Perhaps the functionality for this is already there?
